kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsService.java

```diff
@@ -20,10 +21,26 @@
 @Plugin(configType = Config.class)
 public class VaultKmsService implements KmsService<Config, String, VaultEdek> {
 
+    @SuppressWarnings("java:S3077") // Config is an immutable object
+    private volatile Config config;
+
+    @Override
+    public void initialize(@NonNull Config config) {
+        Objects.requireNonNull(config);
+        if (this.config != null) {
+            throw new IllegalStateException("KMS service is already initialized");
+        }
```

Comment:
Same point as in the AWS case,  I'm not clear why we should enforce this.

Reply:
Ok, removed.